apps/process-new-granules/src/process_new_granules.py

```diff
@@ -1,83 +1,31 @@
-        job_type=subscription['job_specification']['job_type'],
+def handle_subscription_batch(subscriptions: list[dict], cutoff_date: datetime, worker_function_arn: str) -> None:
+    payload = {
+        'subscriptions': subscriptions,
+        'cutoff_date': cutoff_date.isoformat(),
+    }
+    client = boto3.client('lambda')
+    response = client.invoke(
+        FunctionName=worker_function_arn,
+        InvocationType='Event',
+        Payload=payload,
     )
-    processed_granules = [job['job_parameters']['granules'][0] for job in processed_jobs]
-
-    search_results = asf_search.search(**dynamo.util.convert_decimals_to_numbers(subscription['search_parameters']))
-    return [result for result in search_results if result.properties['sceneName'] not in processed_granules]
-
-
-def get_neighbors(granule, depth, platform):
-    stack = asf_search.baseline_search.stack_from_product(granule)
-    stack = [item for item in stack if
-             item.properties['temporalBaseline'] < 0 and item.properties['sceneName'].startswith(platform)]
-    neighbors = [item.properties['sceneName'] for item in stack[-depth:]]
-    return neighbors
-
-
-def get_jobs_for_granule(subscription, granule):
-    job_specification = deepcopy(subscription['job_specification'])
-    if 'job_parameters' not in job_specification:
-        job_specification['job_parameters'] = {}
-    job_specification['subscription_id'] = subscription['subscription_id']
-
-    job_type = job_specification['job_type']
-
-    if job_type in ['RTC_GAMMA', 'WATER_MAP']:
-        job_specification['job_parameters']['granules'] = [granule.properties['sceneName']]
-        payload = [job_specification]
-    elif job_type in ['AUTORIFT', 'INSAR_GAMMA']:
-        payload = []
-        neighbors = get_neighbors(granule, 2, subscription['search_parameters']['platform'])
-        for neighbor in neighbors:
-            job = deepcopy(job_specification)
-            job['job_parameters']['granules'] = [granule.properties['sceneName'], neighbor]
-            payload.append(job)
-    else:
-        raise ValueError(f'Subscription job type {job_type} not supported')
-    return payload
-
-
-def get_jobs_for_subscription(subscription, limit):
-    granules = get_unprocessed_granules(subscription)
-    jobs = []
-    for granule in granules[:limit]:
-        jobs.extend(get_jobs_for_granule(subscription, granule))
-    return jobs
-
-
-def disable_subscription(subscription):
-    subscription['enabled'] = False
-    dynamo.subscriptions.put_subscription(subscription['user_id'], subscription)
-
-
-def handle_subscription(subscription):
-    print(f'Handling subscription {subscription["subscription_id"]} for user {subscription["user_id"]}')
-    jobs = get_jobs_for_subscription(subscription, limit=20)
-    if jobs:
-        print(f'Submitting {len(jobs)} jobs')
-        dynamo.jobs.put_jobs(subscription['user_id'], jobs, fail_when_over_quota=False)
+    # TODO attempt to log worker success or failure?
```

Comment:
`InvocationType='Event'` is a "fire and forget" model. `response` will say "I was able to start the execution", but not "the execution succeeded/failed. I think it's sufficient to log "started worker execution for {subscription_id}" here.